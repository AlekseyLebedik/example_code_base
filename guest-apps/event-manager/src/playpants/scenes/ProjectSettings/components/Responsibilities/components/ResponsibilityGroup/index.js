import React from 'react';
import PropTypes from 'prop-types';

import PermissionSelect from 'dw/core/components/FormFields/PermissionSelect';

import Chip from '@material-ui/core/Chip';
import Paper from '@material-ui/core/Paper';

import styles from './index.module.css';

const SourceResponsibilityChip = ({ index, envType, ...props }) => (
  <div key={`${envType.id}.${index}`}>
    <Chip {...props} />
  </div>
);

SourceResponsibilityChip.propTypes = {
  envType: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired,
};

const ResponsibilityGroup = ({ envType, selection, index }) => (
  <Paper className={styles.container}>
    <label className={styles.titleContent}>{envType.type}</label>
    <>
      <PermissionSelect
        options={envType.responsibilities}
        name={`${selection}.responsibilities`}
        placeholder="Please select Responsibilities"
        label="Responsibilities"
        ItemComponent={SourceResponsibilityChip}
        itemProps={{
          envType,
          index,
        }}
        fullWidth
        multiple
      />
    </>
  </Paper>
);

ResponsibilityGroup.propTypes = {
  envType: PropTypes.object.isRequired,
  selection: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
};

export default ResponsibilityGroup;
