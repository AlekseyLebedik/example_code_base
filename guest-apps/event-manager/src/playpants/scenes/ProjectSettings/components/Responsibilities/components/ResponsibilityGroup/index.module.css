.nestedItem {
  padding-left: 32px;
}

.chips {
  display: flex;
  flex-flow: column wrap;
  margin-top: -12px;
}

.groups {
  display: flex;
  flex-wrap: wrap;
  margin-top: -3px;
}

.groups > div {
  margin-top: 3px;
  margin-right: 3px;
}

.group {
  display: block;
  width: auto;
  max-width: 100%;
  padding: 0;
  margin-bottom: 0px;

  font-weight: 500;
  line-height: inherit;
  color: rgba(0, 0, 0, 0.65);
  white-space: normal;
  text-transform: uppercase;
}

.group {
  margin-top: 6px;
}

.global {
  padding-top: 5px;
  border: 0px;
}

.container {
  padding: 20px;
  margin-top: 10px;
}

.titleContent {
  text-transform: capitalize;

  font-weight: bold;
}
