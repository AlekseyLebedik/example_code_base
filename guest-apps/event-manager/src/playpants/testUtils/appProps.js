import {
  activitySettingsSelector,
  calendarPresetsSettingsSelector,
  colorSettingsSelector,
  currentProjectSelector,
  eventSettingsSelector,
  eventTypeSettingsSelector,
  globalSettingsSelector,
  notificationSettingsSelector,
  platformSettingsSelector,
} from 'playpants/components/App/selectors';
import mockState from './mockState';

export const activitySettings = activitySettingsSelector(mockState);
export const calendarPresetSettings =
  calendarPresetsSettingsSelector(mockState);
export const colorSettings = colorSettingsSelector(mockState);
export const currentProject = currentProjectSelector(mockState);
export const eventSettings = eventSettingsSelector(mockState);
export const eventTypeSettings = eventTypeSettingsSelector(mockState);
export const globalSettings = globalSettingsSelector(mockState);
export const notificationSettings = notificationSettingsSelector(mockState);
export const platformSettings = platformSettingsSelector(mockState);

export const appProps = {
  currentProject,
  fetchProjectSettings: jest.fn(),
  fetchUsers: jest.fn(),
  isMissingSettings: true,
  match: { params: { projectId: 1, env: 'cert' } },
  permissions: {
    adminPermission: true,
    eventWritePermission: true,
    pubvarsWritePermission: true,
    pyscriptWritePermission: true,
    pubstorageWritePermission: true,
    achievementWritePermission: true,
    publisherObjectsWritePermission: true,
    tpDeploymentWritePermission: true,
  },
  setProject: jest.fn(),
  setTitle: jest.fn(),
};
