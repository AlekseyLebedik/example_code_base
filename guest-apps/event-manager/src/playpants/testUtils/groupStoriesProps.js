export const statelessGroupStoriesProps = {
  baseUrl: '',
  currentProject: {},
  eventDetailData: {},
  match: {},
  selectedGroupStory: {},
};

export const groupStoriesDetailProps = {
  currentProject: {},
  eventManagerStoryEventsData: [],
  eventManagerStoryEventsError: {},
  eventManagerStoryEventsLoading: true,
  informationalStoryEventsData: [],
  informationalStoryEventsError: {},
  informationalStoryEventsLoading: true,
  infoTypes: [],
  onFetchEventManagerEvents: jest.fn(),
  onFetchInformationalEvents: jest.fn(),
  onResetGroupStoriesDetail: jest.fn(),
  selectedGroupStory: {},
};

export const groupStoriesSidebarProps = {
  baseUrl: '',
  currentProject: {},
  handleSetSelectedGroupStory: jest.fn(),
  match: { params: { storyId: 1 } },
  onDeleteThenRedirect: jest.fn(),
  onFetchAllGroupStoryEvents: jest.fn(),
  onFetchGroupStoryThenSelect: jest.fn(),
  onResetGroupStoriesSidebar: jest.fn(),
  onSearch: jest.fn(),
  onShowMore: jest.fn(),
  searchedGroupStoriesData: [],
  searchedGroupStoriesLoading: true,
  searchedGroupStoriesNext: '',
  selectedGroupStory: {},
};

export const deleteStoryMessageProps = {
  allGroupStoryEventsData: [],
  openDeletableEventDetail: jest.fn(),
  onSetSelectedThenOpen: jest.fn(),
  onRemoveGroupStoryFromEvent: jest.fn(),
};

export const groupStoryFormDialogProps = {
  action: '',
  currentProject: {},
  form: '',
  handleSubmit: jest.fn(),
  icon: '',
  initialValues: {},
  onCancel: jest.fn(),
  onCreateGroupStory: jest.fn(),
  onPatchGroupStory: jest.fn(),
  title: '',
};

export const groupStoryFormProps = {
  asyncValidating: false,
  onSubmit: jest.fn(),
};

export const listItemProps = {
  currentProject: {},
  item: {},
  onClick: jest.fn(),
  onDeleteThenRedirect: jest.fn(),
  selectedGroupStory: {},
};
