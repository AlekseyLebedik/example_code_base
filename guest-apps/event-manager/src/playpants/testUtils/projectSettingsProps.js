import { groupListSelector } from 'playpants/scenes/ProjectSettings/components/Responsibilities/selectors';

import mockState from './mockState';
import {
  activitySettings,
  colorSettings,
  currentProject,
  eventSettings,
  globalSettings,
  notificationSettings,
  permissions,
} from './appProps';

export const groups = groupListSelector(mockState);

const { ProjectSettings } = mockState.Scenes;
export const users = ProjectSettings.Users.availableUsers.data;

export const projectSettingsProps = {
  ...ProjectSettings,
  permissions,
  baseUrl: '',
  hasCurrentProjectSettings: true,
  match: { params: { id: 1 } },
  updateProjectSetting: jest.fn(),
  onFetchAvailableUsers: jest.fn(),
};

export const statelessProjectSettingsProps = {
  ...projectSettingsProps,
  currentProject,
  currentAuthSettings: [
    {
      title: 1,
      titleName: 'PS3',
      settingID: 4,
      setting: {
        motd: [3],
        playlist: [3],
        pubvars: [3],
        pubstorage: [3],
        pyscript: [3],
        objectStore: [3],
      },
      env: 'Certification',
    },
  ],
  addUserToast: jest.fn(),
  selectedTab: 'responsibilities',
  selectedSubTab: 'users',
  match: {
    params: {
      id: 1,
      category: 'responsibilities',
      subcategory: 'users',
    },
    path: '?category=responsibilities?subcategory=users?id=1',
  },
  baseUrl: '/event-manager/5681/',
};

export const availableUsers = [
  {
    label: 'Admin',
    value: 'Admin',
    id: 3,
    username: 'admin',
    enabled: true,
    isSuperuser: false,
    name: 'Admin',
    email: 'admin@test.com',
    _links: {
      self: {
        href: 'http://127.0.0.1:8081/api/v2/users/3/',
      },
    },
  },
  {
    label: 'Guest',
    value: 'Guest',
    id: 4,
    username: 'guest',
    enabled: true,
    isSuperuser: false,
    name: 'Guest',
    email: 'guest@test.com',
    _links: {
      self: {
        href: 'http://127.0.0.1:8081/api/v2/users/4/',
      },
    },
  },
  {
    label: 'Initial User',
    value: 'Initial User',
    id: 1,
    username: 'username',
    enabled: true,
    isSuperuser: true,
    name: 'Initial User',
    email: 'noreply@demonware.net',
    _links: {
      self: {
        href: 'http://127.0.0.1:8081/api/v2/users/1/',
      },
    },
  },
  {
    label: 'Readonly',
    value: 'Readonly',
    id: 2,
    username: 'readonly',
    enabled: true,
    isSuperuser: false,
    name: 'Readonly',
    email: 'readonly@test.com',
    _links: {
      self: {
        href: 'http://127.0.0.1:8081/api/v2/users/2/',
      },
    },
  },
];

export const responsibilitesAndGamertagsProps = {
  ...projectSettingsProps,
  settingsSubTabs: {
    users: 'Users',
    groups: 'Groups',
  },
  handleTabChange: jest.fn(),
  selectedSubTab: 'users',
  baseUrl: '/event-manager/5681/',
  onFetchGroups: jest.fn(),
  onFetchResponsibilityOptions: jest.fn(),
};

export const statelessGroupsProps = {
  ...projectSettingsProps,
  availableUsers,
  groupList: groups,
  groupLoading: false,
  handleCreateGroup: jest.fn(),
  handleDeleteGroup: jest.fn(),
  handleFetchGroups: jest.fn(),
  onFetchGroups: jest.fn(),
  onSetSelectedGroup: jest.fn(),
  selectedGroup: null,
};

export const statelessUsersProps = {
  ...projectSettingsProps,
  selectedTab: 'Groups',
  onTabChange: jest.fn(),
  selectedItem: {},
};

export const globalSettingsProps = {
  clearSchema: jest.fn(),
  fetchSchema: jest.fn(),
  globalSettings,
  settingSchema: {},
  updateProjectSetting: jest.fn(),
};

export const activitiesSettingsProps = {
  activitySettings,
  clearSchema: jest.fn(),
  fetchSchema: jest.fn(),
  settingSchema: {
    items: {
      required: [
        'type',
        'name',
        'allow_multi_titles',
        'allow_duplication',
        'allow_revert',
        'enabled',
        'context',
      ],
      type: 'object',
      properties: {
        allow_multi_titles: {
          type: 'boolean',
        },
        name: {
          type: 'string',
        },
        enabled: {
          type: 'boolean',
        },
        allow_revert: {
          type: 'boolean',
        },
        allow_duplication: {
          type: 'boolean',
        },
        type: {
          enum: ['ae', 'gvs'],
        },
      },
    },
    $schema: 'http://json-schema.org/draft-07/schema#',
    version: 1,
    type: 'array',
    description: 'Project Setting - Activity Settings',
  },
  updateProjectSetting: jest.fn(),
};

export const eventSettingsProps = {
  eventSettings,
  clearSchema: jest.fn(),
  fetchSchema: jest.fn(),
  settingSchema: {
    description: 'Project Setting - Event Settings',
    version: 1,
    additionalProperties: false,
    $schema: 'http://json-schema.org/draft-07/schema#',
    type: 'object',
    properties: {
      statuses: {
        additionalProperties: false,
        type: 'object',
        properties: {
          scheduled: {
            additionalProperties: false,
            required: ['name', 'is_read_only'],
            type: 'object',
            properties: {
              allow_delete: {
                type: 'boolean',
              },
              is_read_only: {
                type: 'boolean',
              },
              name: {
                type: 'string',
              },
            },
          },
          cancelled: {
            additionalProperties: false,
            required: ['name', 'is_read_only'],
            type: 'object',
            properties: {
              allow_delete: {
                type: 'boolean',
              },
              is_read_only: {
                type: 'boolean',
              },
              name: {
                type: 'string',
              },
            },
          },
        },
      },
    },
  },
  updateProjectSetting: jest.fn(),
};

export const notificationSettingsProps = {
  notificationSettings,
  clearSchema: jest.fn(),
  fetchSchema: jest.fn(),
  settingSchema: {
    $schema: 'http://json-schema.org/draft-07/schema#',
    version: 1,
    type: 'object',
    description: 'Project Setting - Notification Settings',
    properties: {
      email: {
        required: ['address', 'type_settings'],
        type: 'object',
        properties: {
          type_settings: {
            type: 'object',
            properties: {
              event_cancel: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_end: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_open: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_request_approval: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_end_fail: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_publish_now: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_test: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_reject: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_publish_fail: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_end_now: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_approve: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_test_fail: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_end_success: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_publish_success: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_expire: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_schedule: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_publish: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
            },
          },
          address: {
            type: 'string',
          },
        },
      },
      slack: {
        required: ['channel', 'type_settings'],
        type: 'object',
        properties: {
          type_settings: {
            type: 'object',
            properties: {
              event_cancel: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_end: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_open: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_request_approval: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_end_fail: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_publish_now: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_test: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_reject: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_publish_fail: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_end_now: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_approve: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_test_fail: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_end_success: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_publish_success: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_expire: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_schedule: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
              event_publish: {
                additionalProperties: false,
                required: ['enabled'],
                type: 'object',
                properties: {
                  enabled: {
                    type: 'boolean',
                  },
                },
              },
            },
          },
          channel: {
            type: 'string',
          },
        },
      },
    },
  },
  updateProjectSetting: jest.fn(),
};

export const settingSchema = {
  ...projectSettingsProps,
  ...eventSettingsProps,
  formData: eventSettings,
  handleSubmit: jest.fn(),
};

export const statelessSettingSchemaProps = {
  ...settingSchema,
  expandAll: jest.fn(),
  handleSelection: jest.fn(),
  handleChange: jest.fn(),
  handleFormChange: jest.fn(),
  isExpanded: { root: true },
  formContext: {},
  formRef: jest.fn(),
  openKey: ['pending', 'active'],
};

/* Color Settings */
export const colorPickerProps = {
  colorSettings,
  currentProject,
  projectColors: [
    { name: 'test', color: '#000000', title: 1 },
    { name: 'target', color: '#FFFFFF', title: 3 },
  ],
  updateProjectSetting: jest.fn(),
};

export const statelessColorPickerProps = {
  ...colorPickerProps,
  open: false,
  selectedColor: '#000000',
  handleChangeComplete: jest.fn(),
  handleRowSelection: jest.fn(),
  handleAccept: jest.fn(),
  handleCancel: jest.fn(),
};

/* User List */
export const userListProps = {
  ...ProjectSettings,
  userList: users,
  filterGroups: jest.fn(),
  onSearch: jest.fn(),
  onSelectItem: jest.fn(),
  selectedItem: { id: 1 },
  nextPage: 'nextpageurl',
  onShowMore: jest.fn(),
};

export const groupListProps = {
  ...ProjectSettings,
  _createGroupModalLoading: false,
  _createGroupModalVisible: false,
  _createGroupOnRemoteSubmit: jest.fn(),
  _groupModalCancel: jest.fn(),
  _isCreateGroupModalPristine: true,
  _onSearch: jest.fn(),
  _onShowMore: jest.fn(),
  _onSubmit: jest.fn(),
  _openCreateGroupModal: jest.fn(),
  availableUsers,
  deleteGroup: jest.fn(),
  filterGroups: jest.fn(),
  groupList: groups,
  nextGroupsPage: 'nextpageurl',
  onSelectItem: jest.fn(),
  renderExtraFields: jest.fn(),
  selectedItem: {},
  showMore: true,
};

export const listItemProps = {
  connectDragSource: jest.fn(),
  user: { id: 1, username: 'JohnDoe' },
};

/** Responsibility Group */

export const responsibilityGroupProps = {
  envType: {
    id: 2,
    responsibilities: [
      {
        id: 1,
        code: 'authorize_pubstorage',
        name: 'Authorize | Publisher Storage',
      },
      { id: 2, code: 'authorize_motd', name: 'Authorize | MOTD' },
      { id: 3, code: 'authorize_playlist', name: 'Authorize | Playlists' },
      {
        id: 4,
        code: 'authorize_pubvars',
        name: 'Authorize | Publisher Variables',
      },
      { id: 5, code: 'authorize_pyscript', name: 'Authorize | PyScript' },
      {
        id: 6,
        code: 'authorize_objectstore',
        name: 'Authorize | Object Store',
      },
    ],
    type: 'Certification',
  },
  selection: 'envTypes[0]',
  index: 0,
};

/** Responsibility/Gamertag Form */
export const masterDetailFormProps = {
  form: 'GroupsForm',
  formFields: jest.fn(),
};

/** Responsibility/Gamertags Group Details */
export const groupDetailsProps = {
  match: { params: { id: 1 } },
};

export const userDetailsProps = {
  selectedItem: { id: 4, username: 'admin' },
  optionTypes: [],
  selectedTab: 'Groups',
  onTabChange: jest.fn(),
  fetchAssignedGroups: jest.fn(),
  fetchAvailableGroups: jest.fn(),
  addUserToGroup: jest.fn(),
  onSave: jest.fn(),
  assignedGroups: [],
  groups: [],
  isLoading: false,
  initialValuesAdvanced: {},
  initialValuesRespGroups: {},
  project: 1,
  baseUrl: '',
};

export const responsibilityUserGroupsProps = {
  userID: 1,
  assignedGroups: [],
  fetchAssignedGroups: jest.fn(),
  fetchAvailableGroups: jest.fn(),
  addUserToGroup: jest.fn(),
  groups: [
    { id: 10, name: 'group1' },
    { id: 20, name: 'group2' },
  ],
  handleSubmit: jest.fn(() => jest.fn()),
  onSubmit: jest.fn(),
  onReset: jest.fn(),
  classes: {},
  project: 1,
  baseUrl: '',
  initialValues: {},
};

export const formComponentProps = {
  defaultValue: [],
  options: [],
  addUserToGroup: jest.fn(),
  handleSubmit: jest.fn(() => jest.fn()),
  onSubmit: jest.fn(),
  onReset: jest.fn(),
  form: '',
  pristine: true,
  submitting: false,
};

export const usersContainerProps = {
  onFetchUserList: jest.fn(),
  onFetchGroupList: jest.fn(),
  onShowMore: jest.fn(),
  onSearch: jest.fn(),
  renderDetails: jest.fn(),
  selectedItem: {},
  nextPage: null,
  userList: [],
  groups: {},
  project: 1,
  match: { params: { id: 1 } },
  baseUrl: '',
};
