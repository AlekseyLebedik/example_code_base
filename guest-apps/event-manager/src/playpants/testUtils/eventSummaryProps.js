import createStore from 'playpants/store';
import { currentProject } from './appProps';
import mockState from './mockState';

const { store } = createStore();

const { event } = mockState.Scenes.Event;

export const eventSummaryProps = {
  currentProject,
  event,
  setVisible: jest.fn(),
};

export const statelessEventSummaryProps = {
  ...eventSummaryProps,
  store,
  activitiesOnEnd: [],
  activitiesOnStart: [],
  baseUrl: '/event-manager/1/',
  blockPrimaryAction: false,
  classes: {},
  tags: ['test', 'COD'],
  endDate: 'Friday, August 9, 2019 2:47 PM',
  eventDuration: 'Monday, August 5, 2019 3:00 PM - Friday, August 9, 2019',
  eventSummaryTitles: [{ id: 1, name: 'MW-PS3', platform: 'PS3' }],
  publishDate: 'Monday, August 5, 2019 3:00 PM',
  publishTime: '03:00 PM',
  selectedView: 'month',
  uniqOnStart: [],
  uniqOnEnd: [],
  visible: true,
};

export const eventDetailActivityChipsProps = {
  uniqActivities: [],
  activities: [],
  classes: {},
};

export const eventItemChipsProps = {
  classes: {},
  uniqItems: [],
};

export const eventDetailHeaderProps = {
  className: '',
  avatarClassName: '',
  primary: '',
  secondary: '',
};

export const eventDetailItemProps = {
  disableTypography: false,
  icon: '',
  iconClassName: '',
  isHidden: false,
  linkTo: 'test/link',
  primary: '',
  secondary: '',
};

export const statusDotProps = {
  iconClassName: '',
  statusText: '',
};

export const eventDetailConflictProps = {
  ...statelessEventSummaryProps,
  conflicts: [
    {
      conflicts: [
        {
          activity_type: 'pubstorage',
          details: [],
          event_activity: {
            id: 5,
            type: 'pubstorage',
            activity: '{"files":[4]}',
            title_envs: [1],
          },
          overlapping_event_activity: {
            id: 8,
            type: 'pubstorage',
            activity: '{"files":[4]}',
            title_envs: [1],
          },
          severity: 'activity-title-conflict',
          titles: [1],
        },
      ],
      conflicting_event: {
        id: 2,
        title: 'Overlapping Event',
        status: 'open',
        activities: [8],
        project: 1,
      },
      severity: 'activity-title-conflict',
    },
  ],
  eventId: 1,
  onClick: jest.fn(),
  needsConfirmations: { conflictsCheck: true },
  requiresConfirmation: true,
  setNeedsConfirmations: jest.fn(),
};

export const publishNowConfirmationProps = {
  classes: {},
  needsConfirmations: { publishNowCheck: true },
  setNeedsConfirmations: jest.fn(),
};

export const endNowConfirmationProps = {
  classes: {},
  needsConfirmations: { endNowCheck: true },
  setNeedsConfirmations: jest.fn(),
};

export const eventTaskProps = {
  badgeCount: { discussion: 1, tasks: 1 },
  classes: {},
  eventData: { task: { title: 'test', state: 'title state' } },
};

export const eventStatusIndicatorProps = {
  classes: {
    [`eventManager-active-noEnd`]: {
      backgroundColor: `transparent !important`,
    },
    [`eventManager-active`]: {
      backgroundColor: `green`,
    },
  },
  displayTimeWithTitle: true,
  envType: 'Development',
  eventStatus: 'active',
  isCalendarView: true,
  parentClasses: 'test',
  platforms: ['CROSSPLAY', 'PC'],
  taskStatus: 'failed',
  type: 'eventManager',
};
