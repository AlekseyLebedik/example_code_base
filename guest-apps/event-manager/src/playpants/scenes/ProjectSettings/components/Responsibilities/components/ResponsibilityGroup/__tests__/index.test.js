import React from 'react';
import { shallow } from 'enzyme';
import { responsibilityGroupProps as props } from 'playpants/testUtils/projectSettingsProps';

import ResponsibilityGroup from '../index';

describe('ResponsibilityGroup', () => {
  const root = shallow(<ResponsibilityGroup {...props} />);

  it('renders default properly', () => {
    expect(root).toMatchSnapshot();
  });
});
