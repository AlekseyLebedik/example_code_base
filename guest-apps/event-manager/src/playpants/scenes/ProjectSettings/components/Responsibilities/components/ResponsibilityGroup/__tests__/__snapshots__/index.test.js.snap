// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsibilityGroup renders default properly 1`] = `
<WithStyles(ForwardRef(Paper))
  className="container"
>
  <label
    className="titleContent"
  >
    Certification
  </label>
  <PermissionSelect
    ItemComponent={[Function]}
    fullWidth={true}
    itemProps={
      Object {
        "envType": Object {
          "id": 2,
          "responsibilities": Array [
            Object {
              "code": "authorize_pubstorage",
              "id": 1,
              "name": "Authorize | Publisher Storage",
            },
            Object {
              "code": "authorize_motd",
              "id": 2,
              "name": "Authorize | MOTD",
            },
            Object {
              "code": "authorize_playlist",
              "id": 3,
              "name": "Authorize | Playlists",
            },
            Object {
              "code": "authorize_pubvars",
              "id": 4,
              "name": "Authorize | Publisher Variables",
            },
            Object {
              "code": "authorize_pyscript",
              "id": 5,
              "name": "Authorize | PyScript",
            },
            Object {
              "code": "authorize_objectstore",
              "id": 6,
              "name": "Authorize | Object Store",
            },
          ],
          "type": "Certification",
        },
        "index": 0,
      }
    }
    label="Responsibilities"
    multiple={true}
    name="envTypes[0].responsibilities"
    options={
      Array [
        Object {
          "code": "authorize_pubstorage",
          "id": 1,
          "name": "Authorize | Publisher Storage",
        },
        Object {
          "code": "authorize_motd",
          "id": 2,
          "name": "Authorize | MOTD",
        },
        Object {
          "code": "authorize_playlist",
          "id": 3,
          "name": "Authorize | Playlists",
        },
        Object {
          "code": "authorize_pubvars",
          "id": 4,
          "name": "Authorize | Publisher Variables",
        },
        Object {
          "code": "authorize_pyscript",
          "id": 5,
          "name": "Authorize | PyScript",
        },
        Object {
          "code": "authorize_objectstore",
          "id": 6,
          "name": "Authorize | Object Store",
        },
      ]
    }
    placeholder="Please select Responsibilities"
  />
</WithStyles(ForwardRef(Paper))>
`;
